fuzzers/qemu_arm_launcher/src/fuzzer.rs

```diff
@@ -151,7 +154,30 @@ pub fn fuzz() {
         // A fuzzer with feedbacks and a corpus scheduler
         let mut fuzzer = StdFuzzer::new(scheduler, feedback, objective);
 
-        let mut hooks = QemuHooks::new(&emu, tuple_list!(QemuEdgeCoverageHelper::default()));
+        let mut rangemap = RangeMap::<usize, (u16, String)>::new();
+        // Find the end address of the loaded elf
+        let mut max_addr = 0;
```

Comment:
Why not emu::mappings? https://github.com/AFLplusplus/LibAFL/blob/main/libafl_qemu/src/emu.rs#L618

Reply:
I was not aware this exists. Will have a look and try to integrate it

Reply:
The `emu.mappings()` for the `qemu_arm_launcher` look like this:
```
map = 0x010000, 0x0f4000
map = 0x0f4000, 0x104000
map = 0x104000, 0x105000
map = 0x105000, 0x106000
map = 0x106000, 0x129000
map = 0x3fffe000, 0x3ffff000
map = 0x3ffff000, 0x40000000
map = 0x40000000, 0x40001000
map = 0x40001000, 0x40801000
map = 0xffff0000, 0xffff1000
```
Only the first 5 mappings belong to the loaded elf. I think my solution to use the `elf.goblin()` is better because it only looks for the highest address in the symbol table. No magic-numbers needed. Using the mappings I would need to enforce a magic-number to cut-off after the elf section. Let me know what you think.

Reply:
Maybe a bit more details would help:
```
map = [0x010000 - 0x0f4000] ("/home/patrick/Daten/Uni/Master/4.Sem/Masterarbeit/deps/LibAFL_fork/fuzzers/qemu_arm_launcher/libpng_harness")
map = [0x0f4000 - 0x104000] ("")
map = [0x104000 - 0x105000] ("/home/patrick/Daten/Uni/Master/4.Sem/Masterarbeit/deps/LibAFL_fork/fuzzers/qemu_arm_launcher/libpng_harness")
map = [0x105000 - 0x106000] ("/home/patrick/Daten/Uni/Master/4.Sem/Masterarbeit/deps/LibAFL_fork/fuzzers/qemu_arm_launcher/libpng_harness")
map = [0x106000 - 0x129000] ("")
map = [0x3fffe000 - 0x3ffff000] ("")
map = [0x3ffff000 - 0x40000000] ("")
map = [0x40000000 - 0x40001000] ("")
map = [0x40001000 - 0x40801000] ("")
map = [0xffff0000 - 0xffff1000] ("")
```
In brackets there is the file path if known/existing. As our context is to generate a drcov file we only care about mappings that belong to a file. SRAM/Interrupt/... mappings we don't care about. So what I could do is to add all mappings to the drcov module mappings which have a file attached. That would remove the magic-number and add multiple program code files.

Reply:
Drcov does support having multiple module entries for the same file. It is intended for when modules are not contiguous in memory. That could be an option?